### PR TITLE
Add support for scroll index and scroll children to semantics node

### PR DESCRIPTION
**Pull request by jonahwilliams**
It should be theoretically possible to pipe this information through/around scrollable

### CODE REVIEW

**Comment by jonahwilliams** [comment]
This one isn't quite ready, i will make sure my implementation actually works first 🥉

**Comment by jonahwilliams** [comment]
I have this working locally - the only addition is the requirement to exclude hidden children from our lists.

File: shell/platform/android/io/flutter/view/AccessibilityBridge.java [hunk 1]
```diff
@@ -684,6 +684,17 @@ void updateSemantics(ByteBuffer buffer, String[] strings) {
...
                     event.setMaxScrollX((int) max);
                 }
+                event.setFromIndex(object.scrollIndex);
+                event.setItemCount(object.scrollChildren);
+                if (object.childrenInTraversalOrder != null) {
+                    int visibleChildren = 0;
+                    // There may be invisible children at the beginning and end of the list.
+                    for (SemanticsObject child : object.childrenInTraversalOrder) {
+                        if (!child.hasFlag(Flag.IS_HIDDEN))
+                            visibleChildren += 1;
```
Comment (goderbauer) [inline shell/platform/android/io/flutter/view/AccessibilityBridge.java]: Does this produce good results? What if one of the children that's currently visible and included in scrollChildren doesn't contribute semantics? Wouldn't the numbers be misaligned in that case?

**Comment by chinmaygarde** [comment]
Can we land/close this? Just clearing the queue.

**Comment by jonahwilliams** [comment]
Sure, I'm blocked on some design issues on the framework side so I'll close for now.

**Pull request closed without merging**